### DISCUSSION TITLE
[Feature] Decouple Albums on device from backup

### DISCUSSION THREAD

**Discussion by EnochPrime** [immich-app/immich#2117] [Feature Request]
### The feature

On the mobile app in the library tab there is a section titled "Albums on Device". This section is populated with all folders based on the selection from the backup panel. If the user selects only a specific folder to backup then this section will only display one entry and is not very useful. 

I believe it would be better if this section always showed all albums on the device regardless of backup selection.

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by L-Trump** [comment]
According to some discussion on the community, this feature seems to be fine on IOS? So is it possible to implement this feature on Android so that I can choose to upload a single photo from a specific device's album instead of uploading the entire album. This would be very helpful for folders like screenshots. I'd like to upload some useful screenshots rather than stupidly uploading an entire screenshots folder.

**Reply by L-Trump** [reply]
My current solution is to create a "to backup" album on the device. When I need to save a screenshot, I move the image to that album via the system gallery app. But this way, immich becomes a simple backup app instead of a media management tool.

**Comment by nebb00** [comment]
Would love this. One remaining item would remove Google photos

**Comment by Nextross** [comment]
This would be very helpful, as I don't want to upload an entire album (or turn of automatic backups only to select few photos from the album and then turning them on again)

I don't really see a point in having a tab that shows *only* albums selected for uploading

**Comment by LeeseTheFox** [comment]
Totally agree. It seems like a very basic feature to have, I was very surprised to see that this was not implemented/doesn't work correctly in Immich.

**Reply by alextran1502** [reply]
@LeeseTheFox 

It is a problem of fundamental design choices that have changed over time, thus the data models from the initial choice have to change and adapt to the need of the users and, so they create some technical debts. 

The initial choices that fit MY needs were

- No target albums selected for backup, mass upload everything 
- Not showing local assets, only show backup assets
- No support for offline support, everything were fetched by the server

As you see, as the project has grown, we added in support for the "no" and "not" above. The data model of the initial designs basically have limitation. And now to change them, we have to refactor other parts of the code base that are tightly coupled to the mentioned data model, without a full rewrite

I hope it helps explain why such basic feature is not in Immich yet.
